src/lib/notificationWorker.js

```diff
@@ -1,192 +0,0 @@
-      .where('is_deleted', false)
-      .select().then()
-    if (!subscriptions) {
-      return
-    }
-
-    subscriptions = _.values(subscriptions)
-    // log.debug('notifying ' + subscription.owner + ' at ' +
-    //   subscription.target)
-    const self = this
-    const notifications = yield subscriptions.map(function (subscription) {
-      return findOrCreate(self.Notification, {
-        where: {
-          subscription_id: subscription.id,
-          transfer_id: transfer.id
-        },
-        transaction
-      })
-    })
-
-    co(function * () {
-      self.log.debug('emitting transfer-{' + affectedAccounts.join(',') + '}')
-      for (let account of affectedAccounts) {
-        self.emit('transfer-' + account, notificationBody)
-      }
-
-      // We will schedule an immediate attempt to send the notification for
-      // performance in the good case.
-      // Don't schedule the immediate attempt if the worker isn't active, though.
-      if (!self.scheduler.isEnabled()) return
-
-      yield notifications.map(function (notificationAndCreated, i) {
-        const notification = self.Notification.fromDatabaseModel(notificationAndCreated[0])
-        return self.processNotificationWithInstances(notification, transfer, subscriptions[i], fulfillment)
-      })
-      // Schedule any retries.
-      yield self.scheduler.scheduleProcessing()
-    }).catch(function (err) {
-      self.log.warn('immediate notification send failed ' + err.stack)
-    })
-  }
-
-  * processNotification (notification) {
-    notification = this.Notification.fromData(notification)
-    const transfer = yield this.Transfer.findById(notification.transfer_id)
-    const subscription = yield this.Subscription.findById(notification.subscription_id)
-    const fulfillment = yield this.Fulfillment.findByTransfer(transfer.id)
-    yield this.processNotificationWithInstances(notification, transfer, subscription, fulfillment)
-  }
-
-  * processNotificationWithInstances (notification, transfer, subscription, fulfillment) {
-    this.log.debug('sending notification to ' + subscription.target)
-    const subscriptionURI = this.uri.make('subscription', subscription.id)
-    const notificationBody = {
-      id: subscriptionURI + '/notifications/' + notification.id,
-      subscription: subscriptionURI,
-      event: 'transfer.update',
-      resource: transfer.getDataExternal()
-    }
-    if (fulfillment) {
-      if (transfer.state === transferStates.TRANSFER_STATE_EXECUTED) {
-        notificationBody.related_resources = {
-          execution_condition_fulfillment: fulfillment.getDataExternal()
-        }
-      } else if (transfer.state === transferStates.TRANSFER_STATE_REJECTED) {
-        notificationBody.related_resources = {
-          cancellation_condition_fulfillment: fulfillment.getDataExternal()
-        }
-      }
-    }
-    // Sign notification
-    const algorithm = 'CC' // Crypto-condition signatures
```

Comment:
@justmoon can you weigh in on this, since you implemented the signing in the first place?